src/destination/create.ts

```diff
@@ -0,0 +1,53 @@
+import pino from 'pino';
+
+import {
+  DEFAULT_MOCK_OPTIONS,
+  type MockOptions,
+  createStdoutMock,
+} from './mock';
+
+type CreateDestinationOptions = {
+  /**
+   * Whether to mock the destination.
+   *
+   * If this is set to a truthy value, logging calls will be intercepted and
+   * recorded for later inspection.
+   */
+  mock: MockOptions | boolean;
+};
+
+/**
+ * Helper function to create a logging destination.
+ *
+ * If the `mock` parameter is set to a truthy value, logging calls will be
+ * intercepted and recorded for later inspection.
+ *
+ * Returns:
+ *
+ * - A `destination` to be supplied to the second parameter of `createLogger()`
+ * - A `stdoutMock` recorder for testing and troubleshooting logging calls
+ */
+export const createDestination = Object.assign(
+  (opts: CreateDestinationOptions) => {
+    const stdoutMock = createStdoutMock(
+      typeof opts.mock === 'object' ? opts.mock : DEFAULT_MOCK_OPTIONS,
+    );
```

Comment:
There is inherent overhead here in us initialising a `stdoutMock` even in production where `opts.mock: false`. However this should be truly negligible overhead that is only occurred once on startup.